src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java

```diff
@@ -0,0 +1,58 @@
+package com.snowflake.kafka.connector.records;
+
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+
+import java.util.Map;
+
+public class SnowflakeMetadataConfig {
+  boolean createtimeFlag = true;
+  boolean topicFlag = true;
+  boolean offsetAndPartitionFlag = true;
+  boolean allFlag = true;
+
```

Comment:
Since all the variables will be initialized when object created and will not be changed, we can make all of them to be **final**. 
You may need two constructors instead of parseMetadataConfig function.
The default constructor doesn't allow any arguments, and sets all the variables to be true.
The second constructor has a Map<String, String> parameter, and initializes all variables according to this config.

Reply:
ack